### PR TITLE
Create hierarchy change test

### PR DESCRIPTION
**Pull request by furtnerthomas**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -256,28 +257,42 @@ test_integration_evict_ctrl_int_SOURCES = \
     test/integration/main.c
 
 test_integration_sys_initialize_int_LDADD   = $(TESTS_LDADD)
-test_integration_sys_initialize_int_SOURCES = test/integration/sys-initialize.int.c \
+test_integration_sys_initialize_int_SOURCES = \
+	test/integration/sys-initialize.int.c \
```
Comment (tstruk) [inline Makefile.am]: Try to avoid unrelated changes

File: Makefile.am [hunk 2]
```diff
@@ -256,28 +257,42 @@ test_integration_evict_ctrl_int_SOURCES = \
...
     test/integration/main.c
 
 test_integration_start_auth_session_int_LDADD = $(TESTS_LDADD)
-test_integration_start_auth_session_int_SOURCES = test/integration/main.c \
-    test/integration/start-auth-session.int.c
+test_integration_start_auth_session_int_SOURCES = \
+    test/integration/start-auth-session.int.c \
+    test/integration/main.c
+
+
```
Comment (tstruk) [inline Makefile.am]: don't add extra empty lines here

File: test/integration/hierarchy-change-auth.int.c [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+#include "sapi/tpm20.h"
+
+int test_0 (TSS2_SYS_CONTEXT *sapi_context);
+int test_1 (TSS2_SYS_CONTEXT *sapi_context);
+
+int tpm_hierarchies[] = {TPM_RH_OWNER, TPM_RH_PLATFORM};
+// TODO: TPM_RH_LOCKOUT, TPM_RH_ENDORSEMENT do not work in test_1
+
+int
+test_invoke (TSS2_SYS_CONTEXT *sapi_context)
```
Comment (tstruk) [inline test/integration/hierarchy-change-auth.int.c]: Move invoke test to the bottom and remove the forward declaration on test_0 and test_1

File: test/integration/hierarchy-change-auth.int.c [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+	}
+
+	return 0;
+}
+
+/*
+ * Test basic auth value changes for Platform Auth
+ */
+int
+test_0 (TSS2_SYS_CONTEXT *sapi_context)
```
Comment (tstruk) [inline test/integration/hierarchy-change-auth.int.c]: Could you rename test_0 and test_1 to something more meaningful, like test_platform_auth() and test_endorsm_auth()

File: test/integration/hierarchy-change-auth.int.c [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+	sessionData.hmac = resetAuth;
+	// change auth value to different value
+	newAuth.t.size = 0;
+	rval = Tss2_Sys_HierarchyChangeAuth( sapi_context, hierarchy, &sessionsData, &newAuth, 0 );
+	if (rval != TPM_RC_SUCCESS)
+		print_fail("HierarchyChangeAuth FAILED! Response Code : 0x%x", rval);
+
+	return 0;
+}
+
```
Comment (tstruk) [inline test/integration/hierarchy-change-auth.int.c]: Please remove these extra lines

File: test/integration/hierarchy-change-auth.int.c [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+/*
+ * Test basic auth value changes for Platform Auth, Endorsement Auth, Owner Auth
+ */
+int
+test_1 (TSS2_SYS_CONTEXT *sapi_context)
+{
+
```
Comment (tstruk) [inline test/integration/hierarchy-change-auth.int.c]: Declare int j here instead of in the loop itself. C99 will complain about it.

File: test/integration/hierarchy-change-auth.int.c [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+	TPMS_AUTH_COMMAND *sessionDataArray[1];
+
+	sessionDataArray[0] = &sessionData;
+
+	sessionsData.cmdAuths = &sessionDataArray[0];
+
+	print_log("\nHIERARCHY_CHANGE_AUTH TESTS:\n" );
+
+	// Init authHandle
```
Comment (tstruk) [inline test/integration/hierarchy-change-auth.int.c]: Please try to use C style comments  /* ... */

File: test/integration/hierarchy-change-auth.int.c (hunk 5 above)
Comment (furtnerthomas) [reply test/integration/hierarchy-change-auth.int.c]: i've removed the lines

**Comment by tstruk** [comment]
I have cleaned up a bit this PR to make it a single commit.
